ip-utils.c

```diff
@@ -232,23 +200,84 @@ const char *janus_network_address_string_from_buffer(const janus_network_address
 }
 
 int janus_network_lookup_interface(const struct ifaddrs *ifas, const char *iface, janus_network_address *result) {
-	if(result == NULL)
-		return -1;
+	if(ifas == NULL || iface == NULL || result == NULL)
+		return -EINVAL;
 	janus_network_address_nullify(result);
-	if(iface == NULL)
-		return -1;
 	janus_network_query_config q;
 	/* Let's see if iface is an IPv4 address, an IPv6 address, or possibly an interface name */
-	if(janus_network_prepare_device_query(iface,
-			janus_network_query_options_ipv4 | janus_network_query_options_ipv6 | janus_network_query_options_name, &q)) {
+	int res = janus_network_prepare_device_query(iface,
+			janus_network_query_options_ipv4 | janus_network_query_options_ipv6 | janus_network_query_options_name, &q);
+	if(res != 0) {
 		/* None of them..? */
-		return -1;
+		return res;
 	}
 	const struct ifaddrs *found = janus_network_query_devices(ifas, &q);
 	if(!found || janus_network_get_device_address(found, result)) {
 		/* Couldn't find anything on iface */
-		return -1;
+		return -EINVAL;
 	}
 	/* Done */
 	return 0;
 }
+
+int janus_network_detect_local_ip(janus_network_query_options addr_type, janus_network_address *result) {
+	if(result == NULL)
+		return -EINVAL;
+	janus_network_address_nullify(result);
+	gboolean found = FALSE;
+	int fd = -1;
+	if(addr_type == janus_network_query_options_ipv4 || addr_type == janus_network_query_options_any_ip) {
+		/* Let's try IPv4 (FIXME Should probably use other internal methods) */
+		struct sockaddr_in addr;
+		socklen_t len;
+		fd = socket(AF_INET, SOCK_DGRAM, 0);
+		if(fd > -1) {
```

Comment:
Socket is documented to return -1 on error, so it's better to test explicitly for that.